src/itemstackmetadata.cpp

```diff
@@ -1,43 +1,34 @@
 #include "itemstackmetadata.h"
 #include "util/serialize.h"
-#include "util/strfnd.h"
-
-#define DESERIALIZE_START '\x01'
-#define DESERIALIZE_KV_DELIM '\x02'
-#define DESERIALIZE_PAIR_DELIM '\x03'
-#define DESERIALIZE_START_STR "\x01"
-#define DESERIALIZE_KV_DELIM_STR "\x02"
-#define DESERIALIZE_PAIR_DELIM_STR "\x03"
+#include "jsoncpp/json/json.h"
 
 void ItemStackMetadata::serialize(std::ostream &os) const
 {
-	std::ostringstream os2;
-	os2 << DESERIALIZE_START;
+	Json::Value root(Json::objectValue);
 	for (StringMap::const_iterator
 			it = m_stringvars.begin();
 			it != m_stringvars.end(); ++it) {
-		os2 << it->first << DESERIALIZE_KV_DELIM
-		    << it->second << DESERIALIZE_PAIR_DELIM;
+		root[it->first] = it->second;
 	}
-	os << serializeJsonStringIfNeeded(os2.str());
+	Json::FastWriter writer;
+	os << serializeJsonStringIfNeeded(writer.write(root));
 }
 
 void ItemStackMetadata::deSerialize(std::istream &is)
 {
 	std::string in = deSerializeJsonStringIfNeeded(is);
 
 	m_stringvars.clear();
+	m_stringvars[""] = in;
```

Comment:
This breaks this compat code in `LuaItemStack::l_to_table`:
`		if (item.metadata.size() == 1 && item.metadata.contains("")) {`